src/Mod/Sketcher/Gui/EditModeConstraintCoinManager.cpp

```diff
@@ -1311,27 +1330,42 @@ void EditModeConstraintCoinManager::processConstraints(const GeoListFacade& geol
                     assert(Constr->First >= -extGeoCount && Constr->First < intGeoCount);
 
                     Base::Vector3d pnt1(0., 0., 0.), pnt2(0., 0., 0.);
+                    double helperStartAngle = 0.;
+                    double helperRange = 0.;
+
                     if (Constr->First != GeoEnum::GeoUndef) {
                         const Part::Geometry* geo =
                             geolistfacade.getGeometryFromGeoId(Constr->First);
 
                         if (geo->is<Part::GeomArcOfCircle>()) {
-                            const Part::GeomArcOfCircle* arc =
-                                static_cast<const Part::GeomArcOfCircle*>(geo);
+                            auto* arc = static_cast<const Part::GeomArcOfCircle*>(geo);
                             double radius = arc->getRadius();
                             double angle = (double)Constr->LabelPosition;
+                            double startAngle, endAngle;
+                            arc->getRange(startAngle, endAngle, /*emulateCCW=*/true);
                             if (angle == 10) {
-                                double startangle, endangle;
-                                arc->getRange(startangle, endangle, /*emulateCCW=*/true);
-                                angle = (startangle + endangle) / 2;
+                                angle = (startAngle + endAngle) / 2;
+                            }
+                            if (!(angle > startAngle && angle < endAngle)) {
+                                if (angle < startAngle
+                                    && startAngle - angle < angle + 2 * M_PI - endAngle) {
+                                    helperStartAngle = angle;
+                                    helperRange = startAngle - angle;
+                                }
+                                else {
+                                    if (angle < endAngle) {
+                                        angle += 2 * M_PI;
+                                    }
+                                    helperStartAngle = endAngle;
+                                    helperRange = angle - endAngle;
+                                }

```

Comment:
Would it be suitable to move this its own function?

Reply:
Perhaps, though this PR is about restoring what was removed by error. So I'm not sure refactoring belongs to this PR.